azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -970,6 +992,102 @@ private void cleanOlderProjects() {
         }
       }
     }
+
+    private HashSet<Pair<Integer, Integer>> getActiveProjectVersions() {
+      final HashSet<Pair<Integer, Integer>> activeProjectVersions =
+          new HashSet<>();
+      for (final FlowRunner runner : FlowRunnerManager.this.runningFlows.values()) {
+        final ExecutableFlow flow = runner.getExecutableFlow();
+        activeProjectVersions.add(new Pair<>(flow
+            .getProjectId(), flow.getVersion()));
+      }
+      return activeProjectVersions;
+    }
+
+    private boolean isActiveProject(final ProjectVersion version) {
```

Comment:
Do we define active projects as the projects that have currently running flows? What if they have scheduled flows? Should it also be considered as active?

Reply:
Active projects are the one containing currently running flows.